### PR TITLE
Wiki - Create and Assign Custom RBAC Roles - First draft

### PR DESCRIPTION
**Pull request by PaulJohnston88**
## Overview/Summary

Created a doc to explain how to use the module to create new custom RBAC roles and assign them

## This PR adds

1. Documentation on how to use the module to create new custom RBAC roles and assign them
2. Link to article in the ReadMe
3. Link to the article in the sidebar

### Breaking Changes

1. N/A

## Testing Evidence

1. Tested the code in the wiki doc to ensure the role is created and assigned as expected
2. Tested links in the wiki doc, readme and sidebar 

## As part of this Pull Request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/pulls)
- [ ] Associated it with relevant [issues](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/issues), for tracking and closure.
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/tree/main)
- [X] Performed testing and provided evidence.
- [X] Updated relevant and associated documentation.
- [ ] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located in the [Enterprise-Scale repo](https://github.com/Azure/Enterprise-Scale) in the directory: `/docs/wiki/whats-new.md`)

#262

### CODE REVIEW

File: docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+## Overview
+
+This page describes how to use the module to create your own custom RBAC roles and assign them at the appropriate scopes within your Enterprise-Scale deployment.
+
+In this example, we will create a custom RBAC role called "Reader Support Tickets" which will allow members to read everything in a subscription and to also open support tickets.
+
+We will then assign the new RBAC Role at the `es_landingzones` scope.
```
Comment (krowlandson) [inline docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md:7]: `es_landing_zones` :)

File: docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+We will update the built-in configuration with the new custom RBAC role by following these steps:
+
+- Create the role definition file
+
+- Save the new role at the `es_root` scope
+
+- Assign the role at the `es_landingzones` scope
```
Comment (krowlandson) [inline docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md:15]: And again :)

File: docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+## Create Role Definition File
+
+In your `/lib` directory create a `role_definitions` subdirectory if you don't already have one. You can learn more about archetypes and custom libraries in [this article](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/wiki/%5BUser-Guide%5D-Archetype-Definitions).
+
+>NOTE: Creating a `role_definitions` subdirectory is a recommendation only. If you prefer not to create one or to call it something else, the custom roles will still work.
+
+In the `role_definitions` subdirectory, create a `role_definition_es_reader_support_tickets.tmpl.json` file. This file will contain the role definition for our `Reader Support Tickets` role.
+
+To attempt to ensure that every custom role we create has a unique value for "name", we use the uuidv5 function that is built-in to Terraform. You can learn more about uuidv5 [here](https://www.terraform.io/language/functions/uuidv5).
```
Comment (matt-FFFFFF) [inline docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md:33]: Did we consider using Powershell `New-Guid` or `uuidgen` on Linux or macOS?

As long as the UUID is unique it doesn't matter, the `uuidv5()` function is useful when you want a consistent UUID, but in this case it really doesn't matter.

Use of Terraform console requires the working directory to have `terraform init` run, thus creating a barrier.

File: docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+- Assign the role at the `es_landingzones` scope
+
+The files we will use are:
+
+- [lib/role_definitions/role_definition_es_reader_support_tickets.tmpl.json](#librole_definitionsrole_definition_es_reader_support_ticketstmpljson)
+- [lib/archetype_extension_es_root.tmpl.json](#libarchetype_extension_es_roottmpljson)
+- [lib/archetype_extension_es_landing_zones.tmpl.json](#libarchetype_extension_es_landing_zones_tmpljson)
+
+>IMPORTANT: To allow the declaration of custom or expanded templates, you must create a custom library folder within the root module and include the path to this folder using the `library_path` variable within the module configuration. In our example, the directory is `/lib`.
```
Comment (matt-FFFFFF) [inline docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md:23]: Suggest referring to the lib directory as `lib` or `./lib`. Using `/lib` is an absolute path that might confuse people.

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]

File: docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md (hunk 1 above)
Comment (PaulJohnston88) [reply docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md:7]: Updated.

File: docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md (hunk 2 above)
Comment (PaulJohnston88) [reply docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md:15]: Updated.

File: docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md (hunk 4 above)
Comment (PaulJohnston88) [reply docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md:23]: Updated to `lib`.

File: docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md (hunk 3 above)
Comment (PaulJohnston88) [reply docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md]: Updated to include `New-Guid` and `uuidgen` as options.

File: docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+{
+  "extend_es_landing_zones": {
+    "policy_assignments": [],
+    "policy_definitions": [],
+    "policy_set_definitions": [],
+    "role_definitions": [],
+    "archetype_config": {
+      "parameters": {},
+      "access_control": {
+        "[CONTOSO] Reader-Support-Tickets": ["2f449a58-06b8-48a2-b60c-9107be0ca878"]
```
Comment (krowlandson) [inline docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md:120]: Is it worth making it a little clearer that this GUID is the object ID for the group (as mentioned above)?

**Review by krowlandson — CHANGES_REQUESTED** [review]
Looks great, thank you. Just one comment which I think needs to be addressed.

File: docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md (hunk 5 above)
Comment (PaulJohnston88) [reply docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md:120]: Thanks Kevin, I've added a note to remind readers to update the group ObjectID.

**Review by krowlandson — APPROVED** [review]
🚀 Great stuff, thank you!

**Review by matt-FFFFFF — APPROVED** [review]

**Pull request merged**